xdsl/irdl/constraints.py

```diff
@@ -55,7 +56,14 @@ def update(self, other: ConstraintContext):
 
 _AttributeCovT = TypeVar("_AttributeCovT", bound=Attribute, covariant=True)
 
-ResolveType: TypeAlias = Attribute | Sequence[Attribute]
+ResolveType = Attribute | Sequence[Attribute]
+
+T = TypeVar("T")
```

Comment:
Any reason to make this public?
```suggestion
_T = TypeVar("_T")
```